examples/favorites_advanced/redux_favorites_advanced/lib/feature_puppy/edit/ui_components/puppy_edit_form.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import 'package:favorites_advanced_base/core.dart';
+import 'package:favorites_advanced_base/resources.dart';
+
+import '../ui_components/puppy_edit_card.dart';
+import '../views/puppy_edit_view_model.dart';
+import 'puppy_edit_avatar.dart';
+
+class PuppyEditForm extends StatelessWidget {
+  const PuppyEditForm({
+    required GlobalKey<FormState> formKey,
+    required PuppyEditViewModel viewModel,
+    Key? key,
+  })  : _formKey = formKey,
+        _viewModel = viewModel,
+        super(key: key);
+
+  final GlobalKey<FormState> _formKey;
+  final PuppyEditViewModel _viewModel;
+
+  @override
+  Widget build(BuildContext context) => SafeArea(
+          child: SingleChildScrollView(
+        child: Container(
+          padding: const EdgeInsets.all(8),
+          child: Form(
+            key: _formKey,
+            autovalidateMode: _viewModel.isSubmitAttempted
+                ? AutovalidateMode.onUserInteraction
+                : AutovalidateMode.disabled,
+            child: Column(
+              children: [
+                PuppyEditAvatar(
+                  heroTag: '$PuppyCardAnimationTag ${_viewModel.puppy.id}',
+                  imgPath: _viewModel.puppy.asset,
+                  pickImage: (source) => _viewModel.onImagePicker(source!),
+                ),
+                const SizedBox(height: 20),
+                PuppyEditCard(
+                  label: 'Name',
+                  content: TextFormField(
+                    key: const ValueKey('PuppyNameInputField'),
+                    maxLines: 1,
+                    initialValue: _viewModel.puppy.name,
+                    cursorColor: const Color(0xff333333),
+                    style: TextStyles.editableTextStyle,
+                    textInputAction: TextInputAction.next,
+                    maxLengthEnforcement: MaxLengthEnforcement.enforced,
+                    decoration: InputStyles.textFieldDecoration,
+                    onChanged: (name) => _viewModel.onNameChange(name),
+                    validator: (value) => (value == null || value.isEmpty)
```

Comment:
@GoranPrime, please move all validation logic to the business layer

Reply:
@StanevPrime Corrected